### PR TITLE
python3Packages.hydrus-api: init at 4.0.0

### PR DESCRIPTION
**Pull request by D4ndellion**
###### Description of changes

This is a python library for use with the HTTP API exposed by hydrus.

There aren't any unit tests.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by evanjs — APPROVED** [review]

File: pkgs/development/python-modules/hydrus-api/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, requests
+}:
+
+buildPythonPackage rec {
+  pname = "hydrus-api";
+  version = "4.0.0";
```
Comment (fabaff) [inline pkgs/development/python-modules/hydrus-api/default.nix:9]: Upstream says [`Python >= 3.9`](https://gitlab.com/cryzed/hydrus-api/-/blob/master/README.md#requirements).

```suggestion
  version = "4.0.0";
  format = "setuptools";

  disabled = pythonOlder "3.9";
```

File: pkgs/development/python-modules/hydrus-api/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    sha256 = "sha256-4by2TlZJIKElGgaof1w555ik2hUNbg16YekSWwICGmg=";
+  };
+
+  propagatedBuildInputs = [
+    requests
+  ];
+
+  pythonImportsCheck = [ "hydrus_api" ];
+
+  doCheck = false;
```
Comment (fabaff) [inline pkgs/development/python-modules/hydrus-api/default.nix:22]: Please add a comment why the tests are disabled.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 198187` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.hydrus-api</li>
    <li>python39Packages.hydrus-api</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] package builds (see https://github.com/NixOS/nixpkgs/pull/198187#issuecomment-1296155969)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...[nix-shell:~/.cache/nixpkgs-review/pr-198187/results/python310Packages.hydrus-api/lib/python3.10/site-packages]$ python
Python 3.10.7 (main, Sep  5 2022, 13:12:31) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import hydrus_api            
>>> client = hydrus_api.Client()
>>> print(f"Client API version: v{client.VERSION}")
Client API version: v31
```

Looks good to me.

**Pull request merged**